utils/common/src/main/java/org/apache/brooklyn/util/text/Identifiers.java

```diff
@@ -168,9 +168,11 @@ public static String makeRandomPassword(final int length, String... passwordVali
         return Joiner.on("").join(list);
     }
 
+    @SuppressWarnings({ "rawtypes", "unchecked" })  
     protected static String mergeCharacterSets(String... s) {
         Set characters = new HashSet<Character>();
         for (String characterSet : s) {
+            // more efficient to lose the generics here
```

Comment:
What does this comment mean? Given that Java uses erasure (except in very special cases!) then how is it more efficient to lose the generics?

It looks like the generics are wrong: it should be `Set<String>`.